nixos/modules/services/web-apps/miniflux.nix

```
@@ -44,25 +44,53 @@ in
       };
 
       config = mkOption {
-        type =
-          with types;
-          attrsOf (oneOf [
-            str
-            int
-          ]);
-        example = literalExpression ''
-          {
-            CLEANUP_FREQUENCY = 48;
-            LISTEN_ADDR = "localhost:8080";
-          }
-        '';
+        type = types.submodule {
+          freeformType =
+            with types;
+            attrsOf (oneOf [
+              str
+              int
+            ]);
+          options = {
+            LISTEN_ADDR = mkOption {
+              type = types.str;
+              default = "localhost:8080";
+              description = ''
+                Address to listen on. Use absolute path for a Unix socket.
+                Multiple addresses can be specified, separated by commas.
+              '';
+              example = "127.0.0.1:8080, 127.0.0.1:8081";
+            };
+            DATABASE_URL = mkOption {
+              type = types.str;
+              default = "user=miniflux host=/run/postgresql dbname=miniflux";
```

this should only be set when `cfg.createDatabaseLocally` is true

---

Why? There's no harm in having a default that people have to override.

---

If you're not using the local database you'll need to override this anyway.

---

If I use
```suggestion
              default = if cfg.createDatabaseLocally then "user=miniflux host=/run/postgresql dbname=miniflux" else null;
```
then the error message would be
```
error: A definition for option `nodes.default.services.miniflux.config.DATABASE_URL' is not of type `string'. Definition values:
- In `/home/rschuetz/devel/nixpkgs/nixos/modules/services/web-apps/miniflux.nix': null
```
Is that okay? `mkIf` cannot be used here.

---

PTAL